src/scripts/main.js

```diff
@@ -1,5 +1,54 @@
 'use strict';
 
 document.addEventListener('click', e => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const spiderOfW = spider.offsetWidth;
+  const windEl = document.querySelector('.wall');
+  const wind = e.target.closest('.wall');
+  const windElBoLe = windEl.clientLeft;
+  const windElBorderTop = windEl.clientTop;
+  const windElOfW = windEl.offsetWidth;
+  const windElOfL = windEl.offsetLeft;
+  const windElOfT = windEl.offsetTop;
+  const windElOfH = windEl.offsetHeight;
+
+  if (!wind) {
+    return;
+  }
+
+  if (
+    (
+      e.clientX - windElOfL - (spiderOfW / 2 + windElBoLe)
+    ) > (
+      windElOfW - (spiderOfW + windElBoLe * 2)
```

Comment:
It's more effective to move expressions like this to variables instead of moving things like `windElOfT = windEl.offsetTop;` to variables

I'd recommend you declare variables with spider style left and top (coordinates) for the default case (click inside wall) via `let` keyword. And then compare it with max/min available coordinate (if click on the border) and reassign the variable with spider coordinate. That will allows you to get rid from `else if` constructions and code will look better